src/components/stats/statsEventsHandlerSrv.js

```diff
@@ -6,29 +6,35 @@
         function (exerciseEventsConst, StatsSrv, ExerciseTypeEnum, $log, UtilitySrv) {
             var StatsEventsHandlerSrv = {};
 
-            StatsEventsHandlerSrv.addNewExerciseResult = function(exerciseType, exercise, results){
+            StatsEventsHandlerSrv.addNewExerciseResult = function (exerciseType, exercise, results) {
                 return StatsSrv.isExerciseStatsRecorded(exerciseType, exercise.id).then(function (isRecorded) {
                     if (isRecorded) {
                         return;
                     }
 
                     var newStats = {};
+                    var newStat;
 
                     var questionsMap = UtilitySrv.array.convertToMap(exercise.questions);
                     results.questionResults.forEach(function (result) {
                         var question = questionsMap[result.questionId];
-                        var categoryId = question.categoryId;
-
-                        if (isNaN(+categoryId) || categoryId === null) {
-                            $log.error('StatsEventsHandlerSrv: _eventHandler: bad category id for the following question: ', question.id, categoryId);
-                            return;
-                        }
-
-                        if (!newStats[categoryId]) {
-                            newStats[categoryId] = new StatsSrv.BaseStats();
-                        }
-                        var newStat = newStats[categoryId];
-
+                        var categoryIds = {};
+                        categoryIds.categoryId = question.categoryId;
+                        categoryIds.categoryId2 = question.categoryId2;
+                        angular.forEach(categoryIds, function (categoryId) {
+                            if (angular.isDefined(categoryId)) {
+                                if (isNaN(+categoryId) || categoryId === null) {
```

Comment:
This validation (for categoryId) is not correct anymore.
Need to check both categoryId and categoryId2, if both of them are not numbers / undefined, only then it's an error.